types/react-native/index.d.ts

```diff
@@ -3410,6 +3410,166 @@ export interface ImageStatic extends NativeMethodsMixin, React.ComponentClass<Im
 }
 
 /**
+ * @see https://facebook.github.io/react-native/docs/flatlist.html#props
+ */
+export interface FlatListProperties extends React.Props<FlatListStatic> {
+
+    /**
+     * Rendered in between each item, but not at the top or bottom
+     */
+    ItemSeparatorComponent?: React.ComponentClass<any>
+
+    /**
+     * Rendered at the bottom of all the items.
+     */
+    ListFooterComponent?: React.ComponentClass<any>
+
+    /**
+     * Rendered at the top of all the items.
+     */
+    ListHeaderComponent?: React.ComponentClass<any>
+
+    /**
+     * Optional custom style for multi-item rows generated when numColumns > 1
+     */
+    columnWrapperStyle?: ViewStyle
+
+    /**
+     * For simplicity, data is just a plain array. If you want to use something else,
+     * like an immutable list, use the underlying VirtualizedList directly.
+     */
+    data?: any[]
+
+    /**
+     * `getItemLayout` is an optional optimization that lets us skip measurement of dynamic
+     * content if you know the height of items a priori. getItemLayout is the most efficient,
+     * and is easy to use if you have fixed height items, for example:
+     * ```
+     * getItemLayout={(data, index) => (
+     *   {length: ITEM_HEIGHT, offset: ITEM_HEIGHT * index, index}
+     * )}
+     * ```
+     * Remember to include separator length (height or width) in your offset calculation if you specify
+     * `ItemSeparatorComponent`.
+     */
+    getItemLayout?: (data: Array<any>, index: number) => {length: number, offset: number, index: number}
+
+    /**
+     * If true, renders items next to each other horizontally instead of stacked vertically.
+     */
+    horizontal?: boolean
+
+    /**
+     * Used to extract a unique key for a given item at the specified index. Key is used for caching
+     * and as the react key to track item re-ordering. The default extractor checks `item.key`, then
+     * falls back to using the index, like React does.
+     */
+    keyExtractor?: (item: any, index: number) => string
+
+    legacyImplementation?: boolean
+
+    /**
+     * Multiple columns can only be rendered with `horizontal={false}` and will zig-zag like a `flexWrap` layout.
+     * Items should all be the same height - masonry layouts are not supported.
+     */
+    numColumns?: number
```

Comment:
Is this typed as optional because it’s in the `OptionalProps` list in the actual code? In the docs it’s listed as being required. I’m slightly confused about `OptionalProps` myself 🤔